resources/fetching/fetchers/azure/batch_fetcher.go

```diff
@@ -50,21 +52,26 @@ func NewAzureBatchAssetFetcher(log *logp.Logger, ch chan fetching.ResourceInfo,
 func (f *AzureBatchAssetFetcher) Fetch(ctx context.Context, cMetadata fetching.CycleMetadata) error {
 	f.log.Info("Starting AzureBatchAssetFetcher.Fetch")
 
-	for assetType, pair := range AzureBatchAssets {
-		assets, err := f.provider.ListAllAssetTypesByName([]string{assetType})
-		if err != nil {
-			return err
-		}
+	allAssets, err := f.provider.ListAllAssetTypesByName(maps.Keys(AzureBatchAssets))
+	if err != nil {
+		return err
+	}
 
-		if len(assets) == 0 {
-			continue
-		}
+	if len(allAssets) == 0 {
+		return nil
+	}
+
+	assetGroups := lo.GroupBy(allAssets, func(item inventory.AzureAsset) string {
+		return fmt.Sprintf("%s-%s", item.SubscriptionId, item.Type)
```

Comment:
We just need to make sure that we do not have rules that require all resources of a subscription for evaluation (of multiple types grouped).

Reply:
Yes :+1: But I think we will notice that while implementing a rule that has that requirement.